### PR TITLE
Include display server type in "Copy System Info"

### PR DESCRIPTION
**Pull request by Sauermann**
Add info about X11 or Wayland to System Information on LinuxBSD.

### CODE REVIEW

**Review by Calinou — CHANGES_REQUESTED** [review]

**Comment by Calinou** [comment]
Tested locally (rebased on top of `master` 60f3b7967cbd00b4e1f52d33d372646f7bec02f6
), it works as expected.

```
Godot v4.2.dev (38c6a02a8) - Fedora Linux 38 (KDE Plasma) - X 11 - Vulkan (Forward+) - dedicated NVIDIA GeForce RTX 4090 (nvidia; 535.54.03) - 13th Gen Intel(R) Core(TM) i9-13900K (32 Threads)
```

However, `capitalize()` makes it write "X 11" instead of "X11". This could be fixed using `.capitalize().replace(" ", "")`.

File: editor/editor_node.cpp [hunk 1]
```diff
@@ -4376,6 +4376,9 @@ String EditorNode::_get_system_info() const {
 		godot_version += " " + hash;
 	}
 
+#ifdef LINUXBSD_ENABLED
+	const String display_server = OS::get_singleton()->get_environment("XDG_SESSION_TYPE");
```
[the remark this answers is not in the document]
Comment (Sauermann) [reply editor/editor_node.cpp]: Thanks, I have included your change. `capitalize()` makes it write `X 11` (with a whitespace between).

**Comment by Sauermann** [comment]
Yes, that addition fixed it.

**Review by Calinou — APPROVED** [review]

**Review by deralmas — CHANGES_REQUESTED** [review]
I've found one tiny flaw in the logic but other than that and a nitpick this looks great!

File: editor/editor_node.cpp [hunk 2]
```diff
@@ -4376,6 +4376,9 @@ String EditorNode::_get_system_info() const {
 		godot_version += " " + hash;
 	}
 
+#ifdef LINUXBSD_ENABLED
+	const String display_server = OS::get_singleton()->get_environment("XDG_SESSION_TYPE").capitalize().replace(" ", "");
```
[the remark this answers is not in the document]
Comment (Sauermann) [reply editor/editor_node.cpp:4380]: Just to be on the same page ... a few lines down, I check for `display_server.is_empty()`.
In the case, that it is empty, display-server-info is simply left out.
I'm not sure, if adding `Unknown Display Server` would help?!

The comment is a good idea. Will add an explanation.

**Review by deralmas — APPROVED** [review]
Fantastic! This will avoid us so many headaches :)

**Comment by YuriSizov** [comment]
Thanks!

**Comment by YuriSizov** [comment]
Cherry-picked for 4.1.2.

**Pull request merged**